cmd/nerdctl/run_network_linux_test.go

```diff
@@ -513,26 +513,19 @@ func TestSharedNetworkStack(t *testing.T) {
 
 	containerNameJoin := testutil.Identifier(t) + "-nework"
 	defer base.Cmd("rm", "-f", containerNameJoin).AssertOK()
-	cmd := base.Cmd("run",
+	base.Cmd("run",
 		"-d",
 		"--name", containerNameJoin,
 		"--network=container:"+containerName,
 		testutil.CommonImage,
-		"sleep", "infinity")
-	cmd.AssertOK()
-
-	base.Cmd("exec", containerNameJoin, "wget", "-qO-", "http://127.0.0.1:80").
-		AssertOutContains(testutil.NginxAlpineIndexHTMLSnippet)
+		"sleep", "infinity").AssertOK()
 
-	base.Cmd("restart", containerName).AssertOK()
-	base.Cmd("restart", containerNameJoin).AssertOK()
 	base.Cmd("exec", containerNameJoin, "wget", "-qO-", "http://127.0.0.1:80").
 		AssertOutContains(testutil.NginxAlpineIndexHTMLSnippet)
 
 	base.Cmd("restart", containerName).AssertOK()
```

Comment:
guys, I didn't understand why we keep restart here?

Reply:
The context in here: https://github.com/containerd/nerdctl/pull/1214#pullrequestreview-1035897578

Long story short, we share the network through the network namespace(/proc/<pid>/ns/net), if the container is restarted, PID will be changed, here test this case.